StreamChatClient_v3/Controllers/ChannelController_Tests.swift

```diff
@@ -0,0 +1,296 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+@testable import StreamChatClient_v3
+import XCTest
+
+class ChannelController_Tests: StressTestCase {
+    fileprivate var env: TestEnvironment!
+    
+    var client: ChatClient!
+    
+    var channelId: ChannelId!
+    
+    var controller: ChannelController!
+    
+    override func setUp() {
+        super.setUp()
+        
+        env = TestEnvironment()
+        
+        client = Client(config: ChatClientConfig(apiKey: .init(.unique)))
+        
+        channelId = ChannelId(type: .messaging, id: "test")
+        
+        controller = ChannelController(channelQuery: .init(channelId: channelId), client: client, environment: env.environment)
+    }
+    
+    override func tearDown() {
+        weak var weak_env = env
+        weak var weak_client = client
+        weak var weak_controller = controller
+        
+        env = nil
+        client = nil
+        controller = nil
+        
+        // We need to assert asynchronously, because there can be some delegate callbacks happening
+        // on the background queue, that keeps the controller alive, until they have finished.
+        AssertAsync {
+            Assert.willBeNil(weak_env)
+            Assert.willBeNil(weak_client)
+            Assert.willBeNil(weak_controller)
+        }
+        
+        super.tearDown()
+    }
+    
+    func test_clientAndIdAreCorrect() {
+        let controller = client.channelController(for: channelId)
+        XCTAssert(controller.client === client)
+        XCTAssertEqual(controller.channelQuery.cid, channelId)
+    }
+    
+    // MARK: - Start updating tests
+    
+    func test_noChangesAreReported_beforeCallingStartUpdating() throws {
+        // Save a new channel to DB
+        client.databaseContainer.write { session in
+            try session.saveChannel(payload: self.dummyPayload(with: .unique), query: nil)
+        }
+        
+        // Assert the channel is not loaded
+        AssertAsync.staysTrue(controller.channel == nil)
+    }
+    
+    func test_startUpdating_fetchesExistingChannel() throws {
+        // Save two channels to DB (only one matching the query) and wait for completion
```

Comment:
We don't use `query` in the tests at all. Probably some copy&paste leftover?

Reply:
It's a copy/paste leftover but it's intentionally left. There _is_ a query, internally in controller. So the comment is correct: two channels are saved to db, but only one matches the ChannelController query